.github/workflows/phpunit.yml

```
@@ -0,0 +1,66 @@
+name: "PHPUnit tests"
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  phpunit:
+    name: "PHPUnit tests"
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      matrix:
+        dependencies:
+          - "lowest"
+          - "highest"
+          - "locked"
+        php-version:
+          - "7.3"
+          - "7.4"
+          - "8.0"
+        operating-system:
+          - "ubuntu-latest"
+          - "windows-latest"
+          - "macos-latest"
+        include:
+          - php-version: "8.0"
+            dependencies: "highest"
+            operating-system: "ubuntu-latest"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: "pcov"
+          php-version: "${{ matrix.php-version }}"
+          ini-values: memory_limit=-1
+          tools: composer:v2, cs2pr
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v2"
+        with:
+          path: |
+            ~/.composer/cache
+            vendor
+          key: "php-${{ matrix.php-version }}-${{ matrix.dependencies }}"
+          restore-keys: "php-${{ matrix.php-version }}-${{ matrix.dependencies }}"
+
+      - name: "Install lowest dependencies"
+        if: ${{ matrix.dependencies == 'lowest' }}
+        run: "composer update --prefer-lowest --no-interaction --no-progress --no-suggest --ignore-platform-req=php"
+
+      - name: "Install highest dependencies"
+        if: ${{ matrix.dependencies == 'highest' }}
+        run: "composer update --no-interaction --no-progress --no-suggest --ignore-platform-req=php"
+
+      - name: "Install locked dependencies"
+        if: ${{ matrix.dependencies == 'locked' }}
+        run: "composer install --no-interaction --no-progress --no-suggest --ignore-platform-req=php"
```

```suggestion
    - uses: "ramsey/composer-install@v1"
      with:
        dependency-versions: "${{ matrix.dependencies }}"
        composer-options: "--ignore-platform-req=php"
```

For reference: https://github.com/ramsey/composer-install#dependency-versions

Caching is included

---

Is it? Very cool then.

---

I'd rather keep the current setup, which I'm quite familiar with, sorry.

I'll throw in some time to use @ramsey's version at some point in future, but not now.

Also, `--ignore-platform-req=php` should not be used at this point anymore.

---

@sanmai yes it is exactly the same, but much shorter.

@Ocramius 👍  for keeping things in sync one need to change all occurences.